### ISSUE TITLE
Pupil_Capture_v0.5.7_Windows Crashes

### ISSUE THREAD

**Issue by billycysung** [pupil-labs/pupil#193]
Thank you very much for releasing the Pupil_Capture_v0.5.7_Windows and fixing the detection issue of the world camera. However, the new Capture crashes whenever it is open. This is confirmed on two different PCs (Window 7). Can you please look into this?

Thanks!

**Comment by MichaelBarz** [comment]
Hi. In #191 you mentioned that the eye window worked well and only the world window crashed. The world process has some additional tasks to do compared to the eye process. E.g. it creates the settings and recording folder. If there's a folder of an older version this could cause problems.

Try to delete all pupil related folders in `C:\users\<your user>\` as workaround, also the `recording` one. You can backup your recordings by copying them to another place.

Please report if this solved the problem for you.

Best, Michael

**Comment by mkassner** [comment]
Hi @MichaelBarz ,

I can confirm this behaviour. Deleting the rec dir solves the problem. (I tried with the latest hardware.)

This is a bug is windows only and  introduced with: 9053f8e 

I pretty sure it happens in the way I parse the exception here: https://github.com/pupil-labs/pupil/blob/master/pupil_src/capture/recorder.py#L98-L106

On Windows the error message is most likely different and we need to deal with it accordingly.
Can you have a look?

**Comment by Sijben** [comment]
I'm experiencing a similar issue, however the World screen doesn't crash, it simply never opens. Deleted all related directories (including recordings and settings).

**Comment by mkassner** [comment]
Could you post the log files in capture_settings?

BTW: If you switch to Ubuntu none of this is a problem.

**Comment by mkassner** [comment]
@MichaelBarz I attempted to fix. Can you test and bundle?

**Comment by mkassner** [comment]
Should have tested my fix before pushing ...

**Comment by mkassner** [comment]
While I think the issue with the recording dir is fixed with the recent commit. I have replicated @Sijben s error. Its seems that the windows backend is not so good with the new Pupil Camera when used in combination with the 30fps Pupil Camera. Using a HD Pupil Camera Headset (both world and eye are high speed) I saw no problems.

@Sijben please consider using Linux or Mac until we have this fixed. We have tested Pupil Capture with all possible HW combinations and made sure it works. On Windows is a bit of a different story as we currently rely on a different camera capture backend.

**Comment by billycysung** [comment]
Sorry, I am a bit behind of this thread. Confirm that deleting directory sort out the problem of world camera freezing or crashing. For me, everything is working fine on both windows and mac.

Thanks guys!



On 1 Sep 2015, at 6:42 pm, mkassner <notifications@github.com<mailto:notifications@github.com>> wrote:


While I think the issue with the recording dir is fixed with the recent commit. I have replicated @Sijben<https://github.com/Sijben> s error. Its seems that the windows backend is not so good with the new Pupil Camera when used in combination with the 30fps Pupil Camera. Using a HD Pupil Camera Headset (both world and eye are high speed) I saw no problems.

@Sijben<https://github.com/Sijben> please consider using Linux or Mac until we have this fixed. We have tested Pupil Capture with all possible HW combinations and made sure it works. On Windows is a bit of a different story as we currently rely on a different camera capture backend.

**Comment by Sijben** [comment]
I've tried running it on Mac (Mac book pro mid 2012 and a Mac mini 2012), yet here I encounter the same problems as we started with: Both windows open but only the Eye camera shows an image. I'm able to switch between both camera feeds in the eye window. Currently setting up a Ubuntu to try further. (In case anyone wondered, you cannot run pupil on a USB booted OS).

**Comment by mkassner** [comment]
@Sijben it -should- work on mac. Maybe this is a different issue? Could you contact us via email so we can shot circuit this and figure this out?

**Comment by mkassner** [comment]
@billycysung great to hear!

**Comment by mkassner** [comment]
@MichaelBarz will update the windows bundle so that the `recordings` dir issue is fixed this weekend.

**Comment by billycysung** [comment]
Hi,

I just want to add another issue to this post.

So previously, we found that World window crashes every time it boots up and this can be solved by deleting the "recording" and "setting" file. This means every time I boot up pupil capture, I will need to do the deleting steps.

I recently discovered a new issue. If you successfully get Pupil capture to work. However, if you minimise it and open it again, it crashes (can be reproduced every time I minimise the World window)
